gendoc/src/main.rs

```diff
@@ -0,0 +1,123 @@
+//! Generate documentation for a ciphersuite based on another ciphersuite implementation.
+//!
+//! The documentation for each ciphersuite is very similar, with the only difference being
+//! the ciphersuite name.
+//!
+//! To make it easier to update all ciphersuite documentation when a change is needed,
+//! this binary allows updating all of them based on a single one. This binary
+//! uses frost-ristretto255 as the "canonical" one, so:
+//!
+//! - Change any documentation of a public function or struct in `frost-ristretto255/src/lib.rs`
+//! - Run `cargo run --manifest-path gendoc/Cargo.toml` to update the documentation
+//!   of the other ciphersuites.
+
+use std::{fs, iter::zip};
+
+use regex::Regex;
+
+/// Read the public documentation of public symbols in the given file.
+///
+/// This identifiers snippets in the given file with the format:
+///
+/// ```
+/// /// Some documentation
+/// pub [rest of the line...]
+/// ```
+///
+/// It will return details for each match:
+/// - the symbol "name" ("[rest of the line...]" above, but after replacing
+///   any string in `suite_names_code` with "SuiteName")
+/// - the entire documentation string
+/// - the start and end position of the documentation string in the code, which allows
+///   replacing it later
+///
+/// # Parameters
+///
+/// filename: the name of the file to read.
+/// suite_names_code: strings that reference the specific suite in code
+///     inside `fn` and should be ignore when using for replacements.
+///
+/// # Returns
+///
+/// A list with data for each symbol, see above.
+fn read_docs(filename: &str, suite_names_code: &[&str]) -> Vec<(String, String, usize, usize)> {
+    let mut docs = Vec::new();
+    let code = fs::read_to_string(filename).unwrap();
+    let re = Regex::new(r"(?m)((^[ ]*///.*\n)+)\s*pub (.*)").unwrap();
```

Comment:
I'm fine with regex'ing for now as this is an on-demand internal tool - if we are using it a lot, I wonder if there is some library API to hook into rustdoc processing itself, as the regex may be harder to maintain

Reply:
Some time ago I tried this with `syn` but it's a bit annoying, it's not well documented, and I managed just to extract the docs, actually replacing them would be more difficult... so I went with the simplest thing that worked. But yeah, not a fan of regexes myself :sweat_smile: